src/text/read.cc

```diff
@@ -647,8 +647,32 @@ auto ReadFunction(Tokenizer& tokenizer, Context& context) -> OptAt<Function> {
 
 // Section 4: Table
 
-auto ReadLimits(Tokenizer& tokenizer, Context& context) -> OptAt<Limits> {
+auto ReadIndexType(Tokenizer& tokenizer, Context& context) -> At<IndexType> {
   LocationGuard guard{tokenizer};
+  auto token = tokenizer.Peek();
+  if (context.features.memory64_enabled() &&
+      token.type == TokenType::NumericType) {
+    if (token.numeric_type() == NumericType::I32) {
+      tokenizer.Read();
+      return At{token.loc, IndexType::I32};
+    } else if (token.numeric_type() == NumericType::I64) {
+      tokenizer.Read();
+      return At{token.loc, IndexType::I64};
+    }
+  }
+  return At{IndexType::I32};
```

Comment:
Any not return OptAt and return nullopt?

Reply:
Done.